### PR TITLE
[Core] Fix on discontinous mortar mapper

### PR DESCRIPTION
**Pull request by loumalouomega**
@rubenzorrilla, in case of discontinous the inverse map was not build properly

### CODE REVIEW

File: kratos/processes/simple_mortar_mapper_wrapper_process.h [hunk 1]
```diff
@@ -91,11 +91,25 @@ class SimpleMortarMapperProcessWrapper
...
         // Automatic detect the entities
-        if (rOriginModelPart.NumberOfElements() > 0) ThisParameters["origin_are_conditions"].SetBool(false);
-        if (rDestinationModelPart.NumberOfElements() > 0) ThisParameters["destination_are_conditions"].SetBool(false);
+        if (!origin_are_conditions_is_defined) {
+            if (rOriginModelPart.NumberOfElements() > 0) {
+                ThisParameters["origin_are_conditions"].SetBool(false);
+            } else if (rOriginModelPart.NumberOfConditions() == 0) {
+                KRATOS_ERROR << "No conditions defined on origin model part" << std::endl;
+            }
+        }
```
Comment (rubenzorrilla) [inline kratos/processes/simple_mortar_mapper_wrapper_process.h:106]: What happens if my MP has both conditions and elements (e.g. shell)?

Comment (loumalouomega) [reply kratos/processes/simple_mortar_mapper_wrapper_process.h:106]: You should define it by hand, this is the automatic part

Comment (rubenzorrilla) [reply kratos/processes/simple_mortar_mapper_wrapper_process.h:106]: Define what by hand?

File: kratos/processes/simple_mortar_mapper_wrapper_process.h [hunk 2]
```diff
@@ -91,11 +91,25 @@ class SimpleMortarMapperProcessWrapper
     {
         // The default parameters
         Parameters default_parameters = GetDefaultParameters();
+        const bool origin_are_conditions_is_defined = ThisParameters.Has("origin_are_conditions");
```
Comment (loumalouomega) [inline kratos/processes/simple_mortar_mapper_wrapper_process.h:94]: @rubenzorrilla first is check if these Parameters are set

**Comment by loumalouomega** [comment]
@rubenzorrilla This is ready

File: kratos/processes/simple_mortar_mapper_wrapper_process.h (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/processes/simple_mortar_mapper_wrapper_process.h:106]: Well, if you want to keep the this check I'm fine (we assume that the user provides two model parts ready for mapping).  However, I'd throw a warning to inform about the change in the  `origin_are_conditions` from `True` to `False`.

File: kratos/processes/simple_mortar_mapper_wrapper_process.h [hunk 3]
```diff
@@ -91,11 +91,25 @@ class SimpleMortarMapperProcessWrapper
...
+        if (!origin_are_conditions_is_defined) {
+            if (rOriginModelPart.NumberOfElements() > 0) {
+                ThisParameters["origin_are_conditions"].SetBool(false);
+            } else if (rOriginModelPart.NumberOfConditions() == 0) {
+                KRATOS_ERROR << "No conditions defined on origin model part" << std::endl;
+            }
+        }
+        if (!destination_are_conditions_is_defined) {
+            if (rDestinationModelPart.NumberOfElements() > 0) {
+                ThisParameters["destination_are_conditions"].SetBool(false);
```
Comment (rubenzorrilla) [inline kratos/processes/simple_mortar_mapper_wrapper_process.h:109]: Same suggestion in here.

**Comment by loumalouomega** [comment]
Done

**Comment by rubenzorrilla** [comment]
> Done

You misunderstood my point. I was referring to the change in the json. Thus, I'd revert your last commit and commit my suggestion.

File: kratos/processes/simple_mortar_mapper_wrapper_process.h [hunk 4]
```diff
@@ -91,11 +91,25 @@ class SimpleMortarMapperProcessWrapper
...
+        const bool origin_are_conditions_is_defined = ThisParameters.Has("origin_are_conditions");
+        const bool destination_are_conditions_is_defined = ThisParameters.Has("destination_are_conditions");
         ThisParameters.ValidateAndAssignDefaults(default_parameters);
 
         // Automatic detect the entities
-        if (rOriginModelPart.NumberOfElements() > 0) ThisParameters["origin_are_conditions"].SetBool(false);
-        if (rDestinationModelPart.NumberOfElements() > 0) ThisParameters["destination_are_conditions"].SetBool(false);
+        if (!origin_are_conditions_is_defined) {
+            if (rOriginModelPart.NumberOfElements() > 0) {
+                ThisParameters["origin_are_conditions"].SetBool(false);
```
Comment (rubenzorrilla) [inline kratos/processes/simple_mortar_mapper_wrapper_process.h:101]:
```suggestion
                ThisParameters["origin_are_conditions"].SetBool(false);
                KRATOS_WARNING("SimpleMortarMapperProcessWrapper") << "\'origin_are_conditions\' changed to \`False\`. Mapping from elements." << std::endl;
```

File: kratos/processes/simple_mortar_mapper_wrapper_process.h [hunk 5]
```diff
@@ -91,11 +91,25 @@ class SimpleMortarMapperProcessWrapper
...
+            } else if (rOriginModelPart.NumberOfConditions() == 0) {
+                KRATOS_WARNING("SimpleMortarMapperProcessWrapper") << "No conditions defined on origin model part" << std::endl;
+            }
+        }
+        if (!destination_are_conditions_is_defined) {
+            if (rDestinationModelPart.NumberOfElements() > 0) {
+                ThisParameters["destination_are_conditions"].SetBool(false);
```
Comment (rubenzorrilla) [inline kratos/processes/simple_mortar_mapper_wrapper_process.h:109]:
```suggestion
                ThisParameters["destination_are_conditions"].SetBool(false);
                KRATOS_WARNING("SimpleMortarMapperProcessWrapper") << "\'destination_are_conditions\' changed to \`False\`. Mapping to elements." << std::endl;
```

File: kratos/processes/simple_mortar_mapper_wrapper_process.h [hunk 6]
```diff
@@ -91,11 +91,25 @@ class SimpleMortarMapperProcessWrapper
...
+            } else if (rDestinationModelPart.NumberOfConditions() == 0) {
+                KRATOS_WARNING("SimpleMortarMapperProcessWrapper") << "No conditions defined on destination model part" << std::endl;
```
Comment (rubenzorrilla) [inline kratos/processes/simple_mortar_mapper_wrapper_process.h]:
```suggestion
                KRATOS_ERROR << "No conditions defined on destination model part" << std::endl;
```

File: kratos/processes/simple_mortar_mapper_wrapper_process.h [hunk 7]
```diff
@@ -91,11 +91,25 @@ class SimpleMortarMapperProcessWrapper
...
         ThisParameters.ValidateAndAssignDefaults(default_parameters);
 
         // Automatic detect the entities
-        if (rOriginModelPart.NumberOfElements() > 0) ThisParameters["origin_are_conditions"].SetBool(false);
-        if (rDestinationModelPart.NumberOfElements() > 0) ThisParameters["destination_are_conditions"].SetBool(false);
+        if (!origin_are_conditions_is_defined) {
+            if (rOriginModelPart.NumberOfElements() > 0) {
+                ThisParameters["origin_are_conditions"].SetBool(false);
+            } else if (rOriginModelPart.NumberOfConditions() == 0) {
+                KRATOS_WARNING("SimpleMortarMapperProcessWrapper") << "No conditions defined on origin model part" << std::endl;
```
Comment (rubenzorrilla) [inline kratos/processes/simple_mortar_mapper_wrapper_process.h]:
```suggestion
                KRATOS_ERROR << "No conditions defined on origin model part" << std::endl;
```

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
Better user info

**Review by rubenzorrilla — APPROVED** [review]
OK. Thanks for the fix!

**Comment by loumalouomega** [comment]
> OK. Thanks for the fix!

**Comment by loumalouomega** [comment]
> Better user info

imagen

**Pull request merged**